server/bundles/io.cloudbeaver.service.auth/src/io/cloudbeaver/service/auth/ReverseProxyConfigurator.java

```diff
@@ -0,0 +1,96 @@
+package io.cloudbeaver.service.auth;
+
+import io.cloudbeaver.auth.provider.rp.RPAuthProvider;
+import io.cloudbeaver.model.app.WebAppConfiguration;
+import io.cloudbeaver.model.app.WebApplication;
+import io.cloudbeaver.model.app.WebAuthApplication;
+import io.cloudbeaver.model.session.WebSession;
+import io.cloudbeaver.service.DBWServiceServerConfigurator;
+import org.jkiss.code.NotNull;
+import org.jkiss.code.Nullable;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.Log;
+import org.jkiss.dbeaver.model.security.SMAuthProviderCustomConfiguration;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ReverseProxyConfigurator implements DBWServiceServerConfigurator {
+    private static final Log log = Log.getLog(ReverseProxyConfigurator.class);
+
+    @Override
+    public void configureServer(
+        @NotNull WebApplication application,
+        @Nullable WebSession session,
+        @NotNull WebAppConfiguration appConfig
+    ) throws DBException {
+    }
+
+    @Override
+    public void migrateConfigurationIfNeeded(@NotNull WebApplication application) throws DBException {
+        if (migrationNotNeeded(application)) {
+            return;
+        }
+        migrateConfiguration(application);
+    }
+
+    @Override
+    public void reloadConfiguration(@NotNull WebAppConfiguration appConfig) throws DBException {
+
+    }
+
+    private void migrateConfiguration(
+        @NotNull WebApplication application
+    ) {
+        if (!(application instanceof WebAuthApplication authApplication)) {
+            return;
+        }
+
+        SMAuthProviderCustomConfiguration smReverseProxyhProviderConfiguration =
+            authApplication.getAuthConfiguration().getAuthProviderConfiguration(RPAuthProvider.AUTH_PROVIDER);
+        boolean createNew = false;
```

Comment:
redundant flag, all logic can be placed under `smReverseProxyhProviderConfiguration == null`